### PR TITLE
Moving CI to dmake/jenkins

### PR DESCRIPTION
**Pull request by maingoh**
You should now use `dmake shell deepocli` to develop. It uses pyenv to allow to dev/test multiple python version in the same container.
By default the dmake shell will only allow you to test using version `3.6`:
```bash
# This run the tests for 3.6
pytest -vv tests
```
The dev requirements are only installed for 3.6, but if you want you can switch python version using:
```bash
$ pyenv local 3.9.0
# OR
$ export PYENV_VERSION=3.9.0
```
Then the command `pip` and `python` will be set up to use `3.9.0` (you will need to install the dev requirements yourself)
Alternatively, you can test a specific python version in an isolated env using `./deploy/deepomatic-cli/run_test.sh 3.9.0`

Few other changes:
- Tests are run concurrently using parallel
- Speed up tests by using a faster network

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+*.egg-info
```
Comment (thomas-riccardi) [inline .dockerignore:1]: for projects with `dmake.yml` at the root you need extra excludes (docker and git):
- https://github.com/Deepomatic/deepomatic-client-python/blob/master/.dockerignore#L1-L3
- https://github.com/Deepomatic/deepomatic-client-python/blob/master/.gitignore#L1-L2

maybe do a pass on the other excludes too

File: deepomatic/cli/lib/camera.py [hunk 2]
```diff
@@ -114,8 +113,8 @@ def add_start_camera(name, address):
 
 
 def add(args):
-    camera = get_camera_ctrl().add(args.camera_name, args.camera_address)
-    LOGGER.info("Camera added: ".format(camera))
+    get_camera_ctrl().add(args.camera_name, args.camera_address)
+    LOGGER.info("Camera '{}' added".format(args.camera_name))
```
Comment (thomas-riccardi) [inline deepomatic/cli/lib/camera.py:117]: nit: the previous log format was somewhat easier to parse

File: deploy/deepomatic-cli/run_test.sh [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+virtualenv $venv_dirname
+
+# cleanup on error
+trap cleanup EXIT
+
+source $venv_dirname/bin/activate
+pip install -r ${CURRENT_DIR}/requirements.dev.txt
+
+# linter
+flake8 --statistics --verbose deepomatic tests
```
Comment (thomas-riccardi) [inline deploy/deepomatic-cli/run_test.sh:35]: we usually do the flake8 test earlier: in the build step, to get these source errors as fast as possible

File: deploy/deepomatic-cli/install.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+unset PYENV_VERSION
+# prepare first defined python version for development
+pyenv global $python_versions
+pip install -r ${CURRENT_DIR}/requirements.dev.txt
+pip install -e .
+
+find $PYENV_ROOT/versions -type d '(' -name '__pycache__' -o -name 'test' -o -name 'tests' ')' -exec rm -rf '{}' +
+find $PYENV_ROOT/versions -type f '(' -name '*.pyo' -o -name '*.exe' ')' -exec rm -f '{}' +
+
+# Install parallel for concurrent testing
```
Comment (thomas-riccardi) [inline deploy/deepomatic-cli/install.sh:53]: why not use the package from ubuntu?

File: deploy/deepomatic-cli/install.sh [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+                          liblzma-dev \
+                          llvm \
+                          make \
+                          netbase \
+                          pkg-config \
+                          tk-dev \
+                          wget \
+                          xz-utils \
+                          zlib1g-dev \
+                          libgl1-mesa-glx
```
Comment (thomas-riccardi) [inline deploy/deepomatic-cli/install.sh:32]: why do we need this package?

File: deploy/deepomatic-cli/requirements.dev.txt [hunk 6]
```diff
@@ -0,0 +1,5 @@
+flake8==3.8.4
+pytest==4.6.11; python_version <= '2.7'
+pytest==6.1.2; python_version > '2.7'
```
Comment (thomas-riccardi) [inline deploy/deepomatic-cli/requirements.dev.txt:3]: how does it work with newer pip that refuses conflicts?

**Review by thomas-riccardi — APPROVED** [review]
LGTM, modulo the comments

File: README.md [hunk 7]
```diff
@@ -22,6 +22,12 @@ If you need rpc support, prefer:
 pip install deepomatic-cli[rpc]
 ```
 
+# Development
+
+```bash
+dmake shell deepocli
+```
+
```
Comment (thomas-riccardi) [inline README.md:30]: the dev doc is in ./CONTRIBUTING.md

File: tests/test_platform.py [hunk 8]
```diff
@@ -1,22 +1,17 @@
...
 from contextlib import contextmanager
-from utils import modified_environ
+
 
 ROOT = os.path.dirname(os.path.abspath(__file__))
 WORKFLOW_PATH = ROOT + '/workflow.yaml'
 CUSTOM_NODES_PATH = ROOT + '/custom_nodes.py'
 
-# Using specific api key to run this test
-deploy_api_key = os.environ['DEEPOMATIC_DEPLOY_API_KEY']
```
Comment (ibreschi) [inline tests/test_platform.py:11]: Maybe drop the var from env

**Review by ibreschi — APPROVED** [review]
good for me!

File: deploy/deepomatic-cli/install.sh (hunk 5 above)
Comment (maingoh) [reply deploy/deepomatic-cli/install.sh]: opencv needs it (I reduced as much as possible the deps)

File: deploy/deepomatic-cli/install.sh (hunk 4 above)
Comment (maingoh) [reply deploy/deepomatic-cli/install.sh:48]: It is 4 years old .. even in ubuntu 20 ..

File: deepomatic/cli/lib/camera.py (hunk 2 above)
Comment (maingoh) [reply deepomatic/cli/lib/camera.py:117]: True, I was just fixing a flake8 error (and normalizing with other logs). Those this is not very used for now and very experimental (I will need to work back on it soon)

File: deploy/deepomatic-cli/install.sh (hunk 4 above)
Comment (thomas-riccardi) [reply deploy/deepomatic-cli/install.sh:48]: is it an issue for how we use it? gnu projects are often very stable.

File: deploy/deepomatic-cli/run_test.sh (hunk 3 above)
Comment (maingoh) [reply deploy/deepomatic-cli/run_test.sh:35]: In travis it was in the "test" phase as well. Doing it here allows to run it for the various python version (it doesn't detect the same things depending on the python version used to run (install?) it). 
Note however than the build steps (in Dockerfile) doesn't do anything so it is almost as early.

File: deploy/deepomatic-cli/install.sh (hunk 4 above)
Comment (maingoh) [reply deploy/deepomatic-cli/install.sh:48]: I can try, I am usually just not very confident :smile:

File: README.md (hunk 7 above)
Comment (Zanoellia) [reply README.md]: nit: the dev doc can be added to the `.dockerignore`

File: tests/test_platform.py (hunk 8 above)
Comment (maingoh) [reply tests/test_platform.py:11]: It has been :)

File: deploy/deepomatic-cli/install.sh [hunk 9]
```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
```
Comment (Zanoellia) [inline deploy/deepomatic-cli/install.sh:1]: nit: add locales such as ```locale-gen en_US.UTF-8```

File: deploy/deepomatic-cli/run_test.sh (hunk 3 above)
Comment (Zanoellia) [reply deploy/deepomatic-cli/run_test.sh:35]: nit: maybe remove `deepomatic` and `tests` and add dir to exclude in the `setup.cfg`?

**Review by Zanoellia — APPROVED** [review]

File: deploy/deepomatic-cli/run_test.sh (hunk 3 above)
Comment (maingoh) [reply deploy/deepomatic-cli/run_test.sh:35]: I think i had a conflict with some other files but will check !

Comment (thomas-riccardi) [reply deploy/deepomatic-cli/run_test.sh:35]: yes, let's try to keep in mind the goal of `run vanilla commands and it just works` (flake8, pytest, ...)

Comment (maingoh) [reply deploy/deepomatic-cli/run_test.sh]: I improved the setup.cfg it should be fine now

File: tests/utils.py [hunk 10]
```diff
@@ -139,26 +139,26 @@ def init_files_setup():
 
     # Download image and video files
     single_img_pth = download(tmpdir,
-                              'https://s3-eu-west-1.amazonaws.com/deepo-tests/vulcain/images/test.jpg',
+                              'https://storage.googleapis.com/dp-public/tests-deepocli/img.jpg',
```
Comment (thomas-riccardi) [inline tests/utils.py:142]: We have a bucket dedicated for tests resources: `tests-resources.internal.deepomatic.com`; let's try to use that
https://console.cloud.google.com/storage/browser/tests-resources.internal.deepomatic.com

**Pull request merged**